hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -973,6 +974,7 @@ Stream<FileSlice> fetchAllFileSlices(String partitionPath) {
    */
   public Stream<HoodieBaseFile> fetchLatestBaseFiles(final String partitionPath) {
     return fetchAllStoredFileGroups(partitionPath)
+        .filter(fg -> !isFileGroupReplaced(fg))
```

Comment:
Good catch!  I see that `getLatestBaseFiles(String partitionStr)` filters out the replaced file groups.  Should that API be used in Presto Hive connector?  Also, should we audit all similar APIs regarding compaction and clustering?

Still, to be on par with `fetchLatestBaseFiles()`, this needs to be fixed anyway.

Reply:
This is needed when metadata is not enabled in presto-hive connector then we let presto do the listing and just fetch base files using this API after adding to the view. `getLatestBaseFiles(String partitionStr)` does additional work, which we did not want for perf reasons.

> should we audit all similar APIs regarding compaction and clustering?
I did check all other public APIs in this view class. This is the only one.